Libraries/LibDNS/Message.cpp

```diff
@@ -962,6 +1035,24 @@ ErrorOr<Records::SOA> Records::SOA::from_raw(ParseContext& ctx)
     return Records::SOA { move(mname), move(rname), serial, refresh, retry, expire, minimum };
 }
 
+ErrorOr<void> Records::SOA::to_raw(ByteBuffer& buffer) const
+{
+    TRY(mname.to_raw(buffer));
+    TRY(rname.to_raw(buffer));
+
+    auto const output_size = 5 * sizeof(u32);
+    FixedMemoryStream stream { TRY(buffer.get_bytes_for_writing(output_size)) };
+
+    TRY(stream.write_value(static_cast<NetworkOrdered<u32>>(serial)));
+    TRY(stream.write_value(static_cast<NetworkOrdered<u32>>(refresh)));
+    TRY(stream.write_value(static_cast<NetworkOrdered<u32>>(retry)));
+    TRY(stream.write_value(static_cast<NetworkOrdered<u32>>(expire)));
+    TRY(stream.write_value(static_cast<NetworkOrdered<u32>>(minimum)));
```

Comment:
Don't these static_casts just call the normal constructor?
so you could shorten this quite a bit

Reply:
Yeah it calls the ctor, sure.